c++11/include/opentracing/tracer.h

```diff
@@ -0,0 +1,192 @@
+  // incorporate the given StartSpanOption `option_list`.
+  //
+  // A Span with no SpanReference options (e.g., opentracing::ChildOf() or
+  // opentracing::FollowsFrom()) becomes the root of its own trace.
+  //
+  // Examples:
+  //
+  //     opentracing::Tracer& tracer = ...
+  //
+  //     // The root-span case:
+  //     auto span = tracer.StartSpan("GetFeed")
+  //
+  //     // The vanilla child span case:
+  //     auto span = tracer.StartSpan(
+  //         "GetFeed",
+  //         {opentracing::ChildOf(parentSpan.context())})
+  //
+  //     // All the bells and whistles:
+  //     auto span = tracer.StartSpan(
+  //         "GetFeed",
+  //         {opentracing::ChildOf(parentSpan.context()),
+  //         opentracing::Tag{"user_agent", loggedReq.UserAgent},
+  //         opentracing::StartTimestamp(loggedReq.timestamp())})
+  //
+  std::unique_ptr<Span> StartSpan(
+      const std::string& operation_name,
+      std::initializer_list<option_wrapper<StartSpanOption>> option_list = {})
+      const {
+    StartSpanOptions options;
+    options.start_system_timestamp = SystemClock::now();
+    options.start_steady_timestamp = SteadyClock::now();
+    for (const auto& option : option_list) option.get().Apply(options);
+    return StartSpanWithOptions(operation_name, options);
+  }
+
+  virtual std::unique_ptr<Span> StartSpanWithOptions(
+      const std::string& operation_name,
+      const StartSpanOptions& options) const = 0;
+
+  // Inject() takes the `sc` SpanContext instance and injects it for
+  // propagation within `carrier`. The actual type of `carrier` depends on
+  // the value of `format`.
+  //
+  // OpenTracing defines a common set of `format` values (see BuiltinFormat),
+  // and each has an expected carrier type.
+  //
+  // Returns true on success.
+  virtual bool Inject(const SpanContext& sc, CarrierFormat format,
+                      const CarrierWriter& writer) = 0;
+
+  // Extract() returns a SpanContext instance given `format` and `carrier`.
+  //
+  // OpenTracing defines a common set of `format` values (see BuiltinFormat),
+  // and each has an expected carrier type.
+  //
+  // Returns a `SpanContext` that is `non-null` on success.
+  virtual std::unique_ptr<SpanContext> Extract(CarrierFormat format,
+                                               const CarrierReader& reader) = 0;
+
+  // GlobalTracer returns the global tracer.
+  static std::shared_ptr<Tracer> Global();
+
+  // InitGlobalTracer sets the global tracer pointer, returns the
+  // former global tracer value.
+  static std::shared_ptr<Tracer> InitGlobal(std::shared_ptr<Tracer> tracer);
+};
+
+// StartTimestamp is a StartSpanOption that sets an explicit start timestamp for
+// the new Span.
+class StartTimestamp : public StartSpanOption {
+ public:
+  StartTimestamp(SystemTime system_when, SteadyTime steady_when)
+      : system_when_(system_when), steady_when_(steady_when) {}
```

Comment:
Should we also allow you to construct the `StartTimestamp` with a single point in time that then uses that for both the system and steady timestamp values?